extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,639 @@
+    if(size(aero_dust_ext, 1) /= nbin) &
+      error_msg = "aerosol_optics%init(): array aero_dust_ext has the wrong number of size bins"
+    if(size(aero_dust_ext, 2) /= nband) &
+      error_msg = "aerosol_optics%init(): array aero_dust_ext has the wrong number of bands"
+    if(size(aero_salt_ext, 3) /= nband) &
+      error_msg = "aerosol_optics%init(): array aero_salt_ext has the wrong number of bands"
+    if(size(aero_sulf_ext, 2) /= nband) &
+      error_msg = "aerosol_optics%init(): array aero_sulf_ext has the wrong number of bands"
+    if(size(aero_bcar_ext, 1) /= nband) &
+      error_msg = "aerosol_optics%init(): array aero_bcar_ext has the wrong number of bands"
+    if(size(aero_ocar_ext, 1) /= nband) &
+      error_msg = "aerosol_optics%init(): array aero_ocar_ext has the wrong number of bands"
+    if(.not. extents_are(aero_salt_ext, nbin, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_salt_ext isn't consistently sized"
+    if(.not. extents_are(aero_salt_ssa, nbin, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_salt_ssa isn't consistently sized"
+    if(.not. extents_are(aero_salt_asy, nbin, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_salt_asy isn't consistently sized"
+    if(.not. extents_are(aero_sulf_ext, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_sulf_ext isn't consistently sized"
+    if(.not. extents_are(aero_sulf_ssa, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_sulf_ssa isn't consistently sized"
+    if(.not. extents_are(aero_sulf_asy, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_sulf_asy isn't consistently sized"
+    if(.not. extents_are(aero_bcar_rh_ext, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_bcar_rh_ext isn't consistently sized"
+    if(.not. extents_are(aero_bcar_rh_ssa, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_bcar_rh_ssa isn't consistently sized"
+    if(.not. extents_are(aero_bcar_rh_asy, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_bcar_rh_asy isn't consistently sized"
+    if(.not. extents_are(aero_ocar_rh_ext, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_ocar_rh_ext isn't consistently sized"
+    if(.not. extents_are(aero_ocar_rh_ssa, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_ocar_rh_ssa isn't consistently sized"
+    if(.not. extents_are(aero_ocar_rh_asy, nrh, nband)) &
+      error_msg = "aerosol_optics%init(): array aero_ocar_rh_asy isn't consistently sized"
+    if(error_msg /= "") return
+
+    ! Allocate LUT parameters
+    allocate(this%merra_aero_bin_lims(npair,nbin))
+    this%merra_aero_bin_lims = merra_aero_bin_lims
+
+    allocate(this%aero_rh(nrh), &
+             this%rh_nsteps(2), &
+             this%rh_step_sizes(2), &
+             this%rh_offsets(2))
+    this%aero_rh = aero_rh
+    this%rh_nsteps(1) = 17
```

Comment:
@mjiacono The relative humidity discretization should be encoded in the data file and ready from there. No data should be hard-coded.